src/backend/distributed/planner/merge_planner.c

```diff
@@ -959,7 +962,8 @@ DeferErrorIfTargetHasFalseClause(Oid targetRelationId,
 
 		List *baseRestrictionList = relationRestriction->relOptInfo->baserestrictinfo;
 		List *restrictClauseList = get_all_actual_clauses(baseRestrictionList);
-		if (ContainsFalseClause(restrictClauseList))
+		if (ContainsFalseClause(restrictClauseList) ||
```

Comment:
this used to happen on `DeferErrorIfUnsupportedSubqueryPushdown`, now we also do it here as single-shard router queries do not go through here.

Reply:
Onder, I'm still not getting the logic behind this change. This routine `DeferErrorIfTargetHasFalseClause` is specifically looking for a FALSE qual as it will not generate any tasks. I'm not sure how a FALSE JOIN condition would lead to emoty shards for taget?

Reply:
I think  the name of the function is confusing. We pass the target relation's `relationRestriction->relOptInfo->joininfo`, which is `RestrictInfo structures for join clauses involving this rel`.

So, basically, these are filters that are pushed down to the relations based on joins. If the Postgres planner decides to push `where false` to the `target rel`'s joininfo, then we can conclude that the table has `where false`.

I think the name of the function could be `RelationContainsFalseViaJoinInto()` or such